lib/pbio/include/pbdrv/cache.h

```diff
@@ -14,16 +18,35 @@ void pbdrv_cache_prepare_before_dma(const void *buf, size_t sz);
 // by DMA peripherals. This invalidates the relevant cache lines.
 void pbdrv_cache_prepare_after_dma(const void *buf, size_t sz);
 
+#if PBDRV_CONFIG_CACHE_EV3
+
+// The EV3 MMU is configured to additionally map
+// VA 0xDxxxxxxx onto PA 0xCxxxxxxx, covering the DDR RAM region.
+// The 0xD alias is mapped as non-cacheable and non-bufferable
+// and can be used when data needs to be shared with bus-mastering
+// hardware peripherals. The 0xD region is unused in the physical
+// address map and lies at a convenient offset from the 0xC region.
+// The 0xC region, which is the region typically used to access RAM,
+// is cacheable and bufferable.
+#define PBDRV_EV3_UNCACHED_OFFSET   0x10000000
+
 // Accesses a variable via the uncached memory alias
-#define PBDRV_UNCACHED(x)           (*(volatile __typeof__(x) *)((uint32_t)(&(x)) + 0x10000000))
+#define PBDRV_UNCACHED(x)           (*(volatile __typeof__(x) *)((uintptr_t)(&(x)) + PBDRV_EV3_UNCACHED_OFFSET))
 
 // Gets the address of the uncached memory alias of a variable
-#define PBDRV_UNCACHED_ADDR(x)      ((__typeof__(x) *)((uint32_t)(&(x)) + 0x10000000))
+#define PBDRV_UNCACHED_ADDR(x)      ((__typeof__(x) *)((uintptr_t)(&(x)) + PBDRV_EV3_UNCACHED_OFFSET))
 
 // Cache line size
 #define PBDRV_CACHE_LINE_SZ         32
 
 // Align data to a cache line, which is needed for clean RX DMA
-#define PBDRV_DMA_BUF               __attribute__((aligned(PBDRV_CACHE_LINE_SZ), section(".dma")))
+#define PBDRV_DMA_BUF               __attribute__((aligned(PBDRV_CACHE_LINE_SZ), section(".dma,\"aw\",%nobits@")))
```

Comment:
Is "nobits" something we could do in the linker script instead of here?

Reply:
Apparently yes!